src/utils.ml

```diff
@@ -0,0 +1,46 @@
+(* Bind operator *)
+let (>>=) x f = match x with
+  | Some x -> Some (f x)
+  | None -> None
+
+(* Elvis operator ?: *)
+let (|?) x default = match x with
+  | Some x -> x
+  | None -> default
+
+
+let concatl = List.fold_left Filename.concat ""
+let path_of_list = concatl
+
+let rec list_of_path = function
+  | "/" | "." as p -> [p]
+  | p -> (Filename.basename p) :: list_of_path (Filename.dirname p)
+
+let realpath = function
```

Comment:
I think each function in this module should be documented very precisely (in mli probably).